packages/vite/src/node/server/moduleGraph.ts

```diff
@@ -179,20 +182,33 @@ export class ModuleGraph {
     return noLongerImported
   }
 
-  async ensureEntryFromUrl(
-    rawUrl: string,
-    ssr?: boolean,
-    setIsSelfAccepting = true
-  ): Promise<ModuleNode> {
-    const [url, resolvedId, meta] = await this.resolveUrl(rawUrl, ssr)
-    let mod = this.idToModuleMap.get(resolvedId)
+  ensureEntryFromResolved(
```

Comment:
A new method that allows adding a module node without calling `resolveUrl` again.